docs/docs/client-data-fetching.md

```diff
@@ -2,7 +2,219 @@
+
+```javascript:title=gatsby-config.js
+module.exports = {
+  plugins: [
+    {
+      resolve: "gatsby-source-graphql",
+      options: {
+        typeName: "RMAPI",
+        fieldName: "rickAndMorty",
+        url: "https://rickandmortyapi-gql.now.sh/",
+      },
+    },
+  ],
+}
+```
+
+Now we can add the query to our `index.js` page:
+
+```jsx:title=index.js
+import React, { Component } from "react"
+import { graphql } from "gatsby"
+
+// highlight-start
+// This query is executed at build time by Gatsby.
+export const GatsbyQuery = graphql`
+  {
+    rickAndMorty {
+      character(id: 1) {
+        name
+        image
+      }
+    }
+  }
+`
+// highlight-end
+
+class ClientFetchingExample extends Component {
+  render() {
+    // highlight-start
+    const {
+      rickAndMorty: { character },
+    } = this.props.data
+    // highlight-end
+
+    return (
+      <div style={{ textAlign: "center", width: "600px", margin: "50px auto" }}>
+        // highlight-start
+        <h1>{character.name} With His Pupper</h1>
+        <p>Rick & Morty API data loads at build time.</p>
+        <div>
+          <img
+            src={character.image}
+            alt={character.name}
+            style={{ width: 300 }}
+          />
+        </div>
+        // highlight-end
+        <h2>Image of Rick's pupper</h2>
+        <p>This will come from a request on the client</p>
+      </div>
+    )
+  }
+}
+
+export default ClientFetchingExample
+```
+
+## 3. Fetch pupper info and image data on the client
+
+Now we'll finish out by fetching pupper info from the [Dog CEO Dog API](https://dog.ceo/dog-api/). (We'll fetch a random pupper. Rick isn't picky.)
+
+```jsx:title=index.js
+import React, { Component } from "react"
+import { graphql } from "gatsby"
+import axios from "axios" // highlight-line
+
+// This query is executed at build time by Gatsby.
+export const GatsbyQuery = graphql`
+  {
+    rickAndMorty {
+      character(id: 1) {
+        name
+        image
+      }
+    }
+  }
+`
+
+class ClientFetchingExample extends Component {
+  // highlight-start
+  state = {
+    loading: false,
+    error: false,
+    pupper: {
+      img: "",
+      breed: "",
+    },
+  }
+  // highlight-end
+
+  // highlight-start
+  componentDidMount() {
+    this.fetchRicksPupper()
+  }
+  // highlight-end
+
+  render() {
+    const {
+      rickAndMorty: { character },
+    } = this.props.data
+
+    const { img, breed } = this.state.pupper // highlight-line
+
+    return (
+      <div style={{ textAlign: "center", width: "600px", margin: "50px auto" }}>
+        <h1>{character.name} With His Pupper</h1>
+        <p>Rick & Morty API data loads at build time.</p>
+        <p>Dog API data loads at run time.</p> // highlight-line
+        <div>
+          <img
+            src={character.image}
+            alt={character.name}
+            style={{ width: 300 }}
+          />
+        </div>
+        // highlight-start
```

Comment:
Nit, but this is invalid JSX. People sometimes (?) use these docs outside of gatsbyjs.org, so I like to keep these valid, e.g.

```suggestion
         {/* highlight-start */}
```

Reply:
Great point, that didn't occur to me at the time.